File: archives/2017-2018/lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/Converters.java
package ro.ubbcluj.cs.books.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class Converters {
  @TypeConverter
  public Date fromTimestamp(Long value) {
    return value == null ? null : new Date(value);
  }

  @TypeConverter
  public Long dateToTimestamp(Date date) {
    if (date == null) {
      return null;
    } else {
      return date.getTime();
    }
  }
}

File: lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/db/Converters.kt
package ro.ubbcluj.cs.books.db

import android.arch.persistence.room.TypeConverter

import java.util.Date

class Converters {
  @TypeConverter
  fun fromTimestamp(value: Long?): Date? {
    return if (value == null) null else Date(value)
  }

  @TypeConverter
  fun dateToTimestamp(date: Date?): Long? {
    return date?.time
  }
}
